packages/graphql-server/cdk-stack.ts

```diff
@@ -38,9 +39,34 @@ const createStack = () => {
   const handler = 'packages/graphql-server/src/index.graphqlHandler'
   const entrypoint = path.resolve(__dirname, 'bundle.zip')
 
-  const lambdaFunction = createFunction(stack, 'graphql', entrypoint, handler, config)
+  const { ISSUERS, ...env } = config
+
+  const lambdaFunction = createFunction(stack, 'graphql', entrypoint, handler, env)
   const api = createApi(stack, 'api')
-  addLambdaToApi(stack, api, lambdaFunction, { path: '/{proxy+}', method: 'ANY' }, config.COGNITO_USERPOOL_ID)
+
+  const authorizerLambda = createFunction(
+    stack,
+    'graphql-server-authorizer-lambda',
+    path.resolve(__dirname, 'dist', 'authorizer'),
+    'authorizer/index.handler',
+    {
+      ISSUERS: ISSUERS,
+      CLIENT_ID: config.COGNITO_CLIENT_ID,
+      CONNECT_USER_POOL: config.COGNITO_USERPOOL_ID,
+    },
+    undefined,
+    undefined,
+    undefined,
+    aws_lambda.Runtime.NODEJS_18_X,
+  )
+
+  const authorizer = new aws_apigateway.RequestAuthorizer(stack, 'graphql-server-authorizer', {
+    handler: authorizerLambda,
+    identitySources: [aws_apigateway.IdentitySource.header('authorization')],
```

Comment:
We had an issue when looking at the payments service with a possible case sensitivity issue here. The docs don't explicitly call this out for authorizers specifically, but it does mention they are processed in a case sensitive way. See [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-known-issues.html). You might therefore want to build something here to deal with `Authorization` vs `authorization`